File: reto25/reto25.py
def execute(code: str) -> int:
    value = 0
    start_loop = 0
    for i in range(len(code)):
        if code[i] == "+":
            value += 1
        elif code[i] == "-":
            value -= 1
        elif code[i] == ">":
            continue
        elif code[i] == "{":
            if value == 0:
                i = code.index("}", i)
        elif code[i] == "[":
            start_loop = i
            if value == 0:
                i = code.index("]", i)
        elif code[i] == "]":
            if value != 0:
                i = start_loop
        else:
            pass
    return value

File: reto25/reto25.ts
function execute(code: string): number {
  let value = 0;
  let startLoop = 0;
  for (let i = 0; i < code.length; i++) {
    switch (code[i]) {
      case "+":
        value++;
        break;
      case "-":
        value--;
        break;
      case ">":
        continue;
      case "{":
        if (value === 0) i = code.indexOf("}", i);
        break;
      case "[":
        startLoop = i;
        if (value === 0) i = code.indexOf("]", i);
        break;
      case "]":
        if (value !== 0) i = startLoop;
        break;
      default:
        break;
    }
  }
  return value;
}